src/main/java/seedu/address/model/ModelManager.java

```diff
@@ -29,28 +32,75 @@
     private final UserPrefs userPrefs;
     private final FilteredList<Person> filteredPersons;
     private final FilteredList<Event> filteredEvents;
+    private final QueueManager queueManager;
+    private final FilteredList<Room> filteredRooms;
+    private final FilteredList<ReferenceId> filteredReferenceIds;
 
     /**
      * Initializes a ModelManager with the given addressBook and userPrefs.
      */
-    public ModelManager(ReadOnlyUserPrefs userPrefs, ReadOnlyAddressBook addressBook,
-            ReadOnlyAppointmentBook patientSchedule) {
+    public ModelManager(ReadOnlyAddressBook addressBook, ReadOnlyUserPrefs userPrefs, QueueManager queueManager,
+                        ReadOnlyAppointmentBook patientSchedule) {
         super();
-        requireAllNonNull(userPrefs, addressBook, patientSchedule);
-
+        requireAllNonNull(addressBook, userPrefs);
         logger.fine("Initializing with address book: " + addressBook + " and user prefs " + userPrefs);
-
+        this.queueManager = new QueueManager(queueManager);
         this.addressBook = new AddressBook(addressBook);
         this.appointmentBook = new AppointmentBook(patientSchedule);
         this.userPrefs = new UserPrefs(userPrefs);
-        this.filteredPersons = new FilteredList<>(this.addressBook.getPersonList());
-        this.filteredEvents = new FilteredList<>(this.appointmentBook.getEventList());
+        filteredPersons = new FilteredList<>(this.addressBook.getPersonList());
+        filteredRooms = new FilteredList<>(this.queueManager.getRoomList());
+        filteredEvents = new FilteredList<>(this.appointmentBook.getEventList());
+        filteredReferenceIds = new FilteredList<>(this.queueManager.getReferenceIdList());
     }
 
     public ModelManager() {
-        this(new UserPrefs(), new AddressBook(), new AppointmentBook());
+        this(new AddressBook(), new UserPrefs(), new QueueManager(), new AppointmentBook());
+    }
+
+    //=========== QueueManager ==================================================================================
+
+    @Override
+    public QueueManager getQueueManager() {
+        return queueManager;
+    }
+
+    @Override
+    public void removePatient(ReferenceId target) {
```

Comment:
Please give more descriptive names to your functions. Possibility, removeFromQueue(...)